libavcodec/qsvdec.c

```diff
@@ -646,6 +818,13 @@ static int qsv_decode(AVCodecContext *avctx, QSVContext *q,
         return 0;
     }
 
+    ret = get_payload(avctx, q);
```

Comment:
The return code from `MFXVideoDECODE_DecodeFrameAsync` is lost.  We should check whether the return code from `MFXVideoDECODE_DecodeFrameAsync` is `MFX_ERR_MORE_DATA` or `MFX_WRN_VIDEO_PARAM_CHANGED`etc.

Reply:
Thanks for review. I have fixed this issue.